src/server/scripts/Outland/BlackTemple/boss_supremus.cpp

```diff
@@ -178,6 +178,9 @@ class boss_supremus : public CreatureScript
                 default:
                     break;
             }
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
```

Comment:
In this particular case it realy seems to do nothing, because of this:
```cpp
void BossAI::UpdateAI(uint32 diff)
{
    if (!UpdateVictim())
        return;

    events.Update(diff);

    if (me->HasUnitState(UNIT_STATE_CASTING))
        return;

    while (uint32 eventId = events.ExecuteEvent())
    {
        ExecuteEvent(eventId);
        if (me->HasUnitState(UNIT_STATE_CASTING))
            return;
    }

    DoMeleeAttackIfReady();
}